mod/redir.php

```diff
@@ -70,10 +71,9 @@ function redir_init(App $a) {
 				&& is_array($_SESSION['remote']))
 			{
 				foreach ($_SESSION['remote'] as $v) {
-					if (!empty($v['uid']) && !empty($_SESSION['visitor_visiting']) &&
-					    !empty($v['cid']) && !empty($_SESSION['visitor_id']) &&
-					    $v['uid'] == $_SESSION['visitor_visiting'] &&
-					    $v['cid'] == $_SESSION['visitor_id']) {
+					if (!empty($v['uid']) && !empty($v['cid']) &&
+					    $v['uid'] === Session::get('visitor_visiting') &&
```

Comment:
I'm not sure a strict comparison is warranted. In theory it should all be strings so we don't really need to be that strict.